### DISCUSSION TITLE
How to use my own train data

### DISCUSSION THREAD

**Discussion by MechanicalLearningAI** [ultralytics/ultralytics#5205] [Q&A]
Hello everyone. i have a question i already trained my own dataset in roboflow and I want to use my own data to do a Vehicles Tracking, Counting (Entering & Leaving) and Speed Estimation. how I can do that. thank you

**Comment by glenn-jocher** [comment]
@MechanicalLearningAI hello! Great to hear that you've trained your own dataset. For vehicle tracking, counting, and speed estimation using YOLOv8, you'll want to follow these steps:

1. **Tracking**: Use the Track mode with your trained model to track vehicles in real-time. This will give you the trajectories of the vehicles across frames.

2. **Counting**: To count vehicles entering and leaving, you'll need to define zones or lines in the field of view and increment counts based on the direction of the tracked vehicle trajectories crossing these lines.

3. **Speed Estimation**: For speed estimation, you'll need to calibrate your camera setup to relate pixels to real-world units (like meters). Then, you can calculate the speed based on the change in position over time of the tracked vehicles.

For more detailed guidance on each mode and task, please refer to the Ultralytics Docs. The Track mode documentation will be particularly useful for getting started with real-time object tracking. Good luck with your project! 🚗💨🔢
